### PR TITLE
Feature: Recurrent Anakin IPPO

### PR DESCRIPTION
**Pull request by DriesSmit**
## What?
Add a recurrent IPPO example.
## Why?
Add another system type.
## How?
Added a file called `ippo_recurrent_anakin.py`.
## Extra
Requires that #902 is merged.

### CODE REVIEW

File: jax_distribution/systems/ippo_recurrent_anakin.py [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+                    "returned_episode_returns": ep_returns,
+                    "returned_episode_lengths": ep_lengths,
+                    "returned_episode": ep_done,
+                }
+
+            transition = Transition(
+                done, action, value, reward, log_prob, last_obs, 
+                info,
+            )
+            runner_state = (train_state, env_state, obsv, done, hstate, rng)
```
Comment (DriesSmit) [inline jax_distribution/systems/ippo_recurrent_anakin.py:148]: Turn this observation to a timestep. This is required if we want to add masking in the network.

File: jax_distribution/systems/ippo_recurrent_anakin.py [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+
+        actor_mean = nn.Dense(128, kernel_init=orthogonal(2), bias_init=constant(0.0))(
+            embedding
+        )
+        actor_mean = nn.relu(actor_mean)
+        actor_mean = nn.Dense(
+            self.action_dim, kernel_init=orthogonal(0.01), bias_init=constant(0.0)
+        )(actor_mean)
+
+        pi = distrax.Categorical(logits=actor_mean)
```
Comment (DriesSmit) [inline jax_distribution/systems/ippo_recurrent_anakin.py:83]: Add masking here.

File: jax_distribution/systems/ippo_recurrent_anakin.py [hunk 3]
```diff
@@ -0,0 +1,440 @@
...
+    rng, _rng = jax.random.split(rng)
+    init_x = (
+        process_observation(env.observation_spec().generate_value(), env_name),
+        jnp.zeros((1)),
+    )
+    init_hstate = ScannedRNN.initialize_carry(config["NUM_AGENTS"], 128)
+    
+    # Support for multiple agents
+    init_net_x, init_net_hstate = jax.tree_util.tree_map(
+        lambda x: x[None, ...],
```
Comment (DriesSmit) [inline jax_distribution/systems/ippo_recurrent_anakin.py:356]: Why is this extra batch dimension needed? It is also used in the ff case. Is the observation not already batched by agents?

**Comment by RuanJohn** [comment]
Closing for now in favour of newer IPPO code.
